miasm2/core/graph.py

```diff
@@ -229,3 +229,38 @@ def compute_postdominators(self, leaf):
                                                 self.reachable_parents,
                                                 self.successors_iter,
                                                 self.predecessors_iter)
+
+    def compute_immediate_dominators(self, head):
+        """Compute the immediate dominators of the graph"""
+        dominators = self.compute_dominators(head)
+        idoms = {}
+
+        for n in dominators:
+            for p in self.reachable_parents(n):
```

Comment:
Since reachable_parents relies on set(), I am not sure it gives any guarantee on the order of the iteration on the reachable parents (i.e., the first yielded may not be the closest to the node). Therefore I think that in some cases, the first dominator-reachable-parent encountered will not be the closest to the node.
I may be unclear, but do you see what I mean?